eastlake/steps/newish_metacal/metacal/metacal_fitter.py

```diff
@@ -125,14 +112,14 @@ def _do_all_metacal(self, mbobs_list, data=None):
             passed_flags, mbobs = self._check_flags(_mbobs)
             if passed_flags:
                 try:
-                    if not NGMIX_V1:
-                        _fit_all_psfs([mbobs], self['metacal']['psf'])
-
-                    boot = self._do_one_metacal(mbobs)
-                    if isinstance(boot, dict):
-                        res = boot
+                    if NGMIX_V2:
+                        res = self._do_one_metacal_ngmixv2(mbobs)
                     else:
-                        res = boot.get_metacal_result()
+                        boot = self._do_one_metacal_ngmixv1(mbobs)
+                        if isinstance(boot, dict):
+                            res = boot
+                        else:
+                            res = boot.get_metacal_result()
                 except (BootPSFFailure, BootGalFailure) as err:
```

Comment:
I am catching any errors raised from the fitters here, including `BootPSFFailure`.